### PR TITLE
[test] modify MOE tests to use NCCL

### PR DESCRIPTION
**Pull request by msbaines**
NCCL all_to_all is now supported in PyTorch (since v1.8.0)

Fixes: #548

# Before submitting

- [ ] Was this discussed/approved via a Github issue? (no need for typos, doc improvements)
- [ ] Did you read the [contributor guideline](https://github.com/facebookresearch/fairscale/blob/master/CONTRIBUTING.md)?
- [ ] Did you make sure to update the docs?
- [ ] Did you write any new necessary tests?

## What does this PR do?
Fixes # (issue).

## PR review
Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

**Comment by msbaines** [comment]
ping

File: tests/nn/moe/test_moe_layer.py [hunk 1]
```diff
@@ -3,53 +3,59 @@
...
 from fairscale.nn import MOELayer, Top2Gate
+from fairscale.utils.testing import torch_version
 
-skip_if_no_cuda = pytest.mark.skipif(not torch.cuda.is_available(), reason="cuda required")
+pytestmark = pytest.mark.skipif(
+    not (torch.cuda.is_available() and torch_version() >= (1, 8, 0)), reason="cuda and torch>=1.8.0 required"
+)
 
-BACKEND = dist.Backend.NCCL if torch.cuda.is_available() else dist.Backend.GLOO  # type: ignore
+devices = ["cuda"]
```
Comment (anj-s) [inline tests/nn/moe/test_moe_layer.py:21]: why don't we want cpu anymore?

Comment (anj-s) [reply tests/nn/moe/test_moe_layer.py:21]: i just read the linked issue. Guessing CPU was a stop gap till all_to_all was supported?

**Review by anj-s — APPROVED** [review]

File: tests/nn/moe/test_moe_layer.py (hunk 1 above)
Comment (msbaines) [reply tests/nn/moe/test_moe_layer.py:21]: NCCL does not support CPU tensors.

**Pull request merged**